src/resources/SearchAnalysis/SearchAnalysisInterfaces.ts

```diff
@@ -0,0 +1,143 @@
+import {PostSearchBodyQueryParams} from '../Search/SearchInterfaces';
+
+type RequestParameters = PostSearchBodyQueryParams;
+interface Result {
+    title: string;
+    rankingInfo: unknown;
+    fields: Record<string, string>;
+}
+
+interface ExecutionStep {
+    type: string;
+    name: string;
+}
+
+interface Condition {
+    id: string;
+}
+
+interface AffectedRequestParameters {
+    affectedRequestParameters: Array<{name: string; value: unknown}>;
+}
+
+interface AffectedResultsPosition {
+    affectedResultsPosition: number[];
+}
+
+interface AppliedRules<Affected = Record<string, unknown>, OtherPropertiesApplied = Record<string, unknown>> {
+    appliedCount: number;
+    applied: Array<
+        {
+            id: string;
+            condition?: Condition;
+        } & Affected &
+            OtherPropertiesApplied
+    >;
```

Comment:
In general, I prefer to avoid nesting objects in definitions like this and instead have more interfaces. That way, if a user needs to build that portion of the interface independently, they don't need to do something like:
```ts
getApplied(): AppliedRules["applied"][number] {
  ...
}
```

I don't know if this is usually the case in this library though.